### PR TITLE
bpo-10483: http.server - what is executable on Windows

### PR DESCRIPTION
**Pull request by MaxwellDupre**
mikecmcleod

https://bugs.python.org/issue10483

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@MaxwellDupre

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Lib/http/server.py [hunk 1]
```diff
@@ -1033,6 +1033,17 @@ def is_python(self, path):
         head, tail = os.path.splitext(path)
         return tail.lower() in (".py", ".pyw")
 
+    def make_cmdline( self, scriptfile, query ):
+        cmdline = [scriptfile]
+        if self.is_python(scriptfile):
+            interp = sys.executable
+            if interp.lower().endswith("w.exe"):
+                # On Windows, use python.exe, not pythonw.exe
+                interp = interp[:-5] + interp[-4:]
```
Comment (eryksun) [inline Lib/http/server.py]: This check isn't necessary. The child process is spawned using pipes for standard I/O, which works fine with the non-console "pythonw.exe" executable. This is probably left over from when it called `os.popen2()` ([link](https://github.com/python/cpython/commit/e7d6b0a22e07adfa33cfa8658acf4c0b3cbecb39#diff-590102ca60c804a5ea71a9a5e03b0ae0b5806be5f2452e2ca726cdeb6092558fR216-R234) to Guido's revision that added this code 21 years ago). 

Even if it did matter, it should only check for exactly "pythonw.exe" nowadays, e.g. `os.path.normcase(os.path.basename(interp)) == os.path.normcase("pythonw.exe")`.

**Comment by MaxwellDupre** [comment]
Hi Eryk,

Should I remove it from the change? If yes, how do I do that?
Revert/cancel and resubmit another change/pull? Or another change and
squash the two?

Regards,
Mike

**Comment by ethanfurman** [comment]
Mike,  make the change in your repo, then push the PR to github again.

File: Lib/modulefinder.py [hunk 2]
```diff
@@ -113,17 +113,19 @@ def __repr__(self):
...
         self.path = path
         self.modules = {}
         self.badmodules = {}
         self.debug = debug
+        self.recurse = 0
         self.indent = 0
         self.excludes = excludes if excludes is not None else []
         self.replace_paths = replace_paths if replace_paths is not None else []
         self.processed_paths = []   # Used in debugging only
+        self.recurse = recurse
```
Comment (ethanfurman) [inline Lib/modulefinder.py:129]: no need to set `self.recurse` twice

File: issue1284670.patch [hunk 3]
```diff
@@ -0,0 +1,171 @@
+--- a/Lib/modulefinder.py
```
Comment (ethanfurman) [inline issue1284670.patch]: the `issue1284670.patch`  file should not be part of the PR, and I'm guessing that `_bootstrap_python` should also not be part of the PR

**Review by ethanfurman — CHANGES_REQUESTED** [review]

File: Lib/modulefinder.py (hunk 2 above)
Comment (MaxwellDupre) [reply Lib/modulefinder.py:129]: I have made the change in local repro.

File: issue1284670.patch (hunk 3 above)
Comment (MaxwellDupre) [reply issue1284670.patch]: I have tried to delete in my Github copy but I can only delete _bootstrap_python. Github doesn't allow me to delete the patch file in my cpython copy. How do I get rid of it?

File: Lib/http/server.py (hunk 1 above)
Comment (MaxwellDupre) [reply Lib/http/server.py]: I have made the change in local pushed to Github but I cant figure out how to push it to Python/cpython from my copy, can you give me a clue?

**Comment by arhadthedev** [comment]
Please add a NEWS entry (you can use https://blurb-it.herokuapp.com/ for it).

And you also included `_bootstrap_python` by mistake.

**Comment by MaxwellDupre** [comment]
Sorry for the mess. Its better I close this PR and I will submit a cleaner PR later. Thanks for your comments.
